evm-contracts/src/v0.6/PreCoordinator.sol

```diff
@@ -229,22 +210,23 @@ contract PreCoordinator is ChainlinkClient, Ownable, ChainlinkRequestInterface,
     recordChainlinkFulfillment(_requestId)
     returns (bool)
   {
-    uint256 minResponses = serviceAgreements[serviceAgreementRequests[_requestId]].minResponses;
+    ServiceAgreement memory sa = serviceAgreements[serviceAgreementRequests[_requestId]];
     bytes32 cbRequestId = requests[_requestId];
     bytes32 saId = serviceAgreementRequests[_requestId];
     delete requests[_requestId];
     delete serviceAgreementRequests[_requestId];
     emit ServiceAgreementResponseReceived(saId, cbRequestId, msg.sender, _data);
-    if (requesters[cbRequestId].responses.push(_data) == minResponses) {
-      serviceAgreements[saId].activeRequests = serviceAgreements[saId].activeRequests.sub(1);
-      Requester memory req = requesters[cbRequestId];
-      delete requesters[cbRequestId];
+    requesters[cbRequestId].responses.push(_data);
+    Requester memory req = requesters[cbRequestId];
+    if (req.responses.length == sa.minResponses) {
       int256 result = Median.calculate(req.responses);
       emit ServiceAgreementAnswerUpdated(saId, cbRequestId, result);
       // solhint-disable-next-line avoid-low-level-calls
       (bool success, ) = req.callbackAddress.call(abi.encodeWithSelector(req.callbackFunctionId, cbRequestId, result));
       return success;
     }
+
+    if (req.responses.length == sa.oracles.length) delete requesters[cbRequestId];
```

Comment:
If `sa.oracles.length == sa.minResponses` this will never get executed because we'll return on 228.

Reply:
It will get executed because later answers will be greater than `sa.minResponses` and that if condition only executed if equal.

Reply:
Right, but if minResponses is equal to length there won't be any later answers.

Reply:
:man_facepalming: I get it now.